### ISSUE TITLE
[REQUEST] Implement a `targeting` mechanism.  

### ISSUE THREAD

**Issue by zprobst** [nodestream-proj/nodestream#138]
**Is your feature request related to a problem? Please describe.**

Ideally it would be possible to build pipelines irrespective of the write configuration applied. Ideally we don't need to pass to include a `GraphDatabaseWriter` to the API, instead our `nodestream.yaml` file can house database configurations and we can target a particular database when calling `run`.  This approach has the following advantages:

1. Allows pipelines to de developed independent of the database target. 
2. Enables plugins to "hook into" the targeting api for more complex features. 
3. Significantly reduces the need for plugins to rely on config for databases only. Allows plugins to be decoupled from db provider.

**Describe the solution you'd like**
**Step 1** - Introduce a `targets` section in the `nodestream.yaml`
 file:

```yaml
# nodestream.yaml
# scope, config sections omitted
targets:
   my-db:
      database: neo4j 
      uri: localhost:7687 
      username: neo4j
      password: neo4j 
```

Note, for this example i've hard coded values, but these should work the way `config` blocks do where it accepts reading from environment and other argument resolvers. 

**Step 2** - Remove the requirement to add a database writer.

```diff
- - implementation: nodestream.databases:GraphDatabaseWriter
-  arguments:
-    batch_size: 1000
-    database: !config neo4j_database
-   uri: !config neo4j_uri
-    username: !config neo4j_username
-    password: !config neo4j_password
```

Note: I think it should still be _allowed_ to add this block, but no longer required. 

Step 3 - Specify targets at run time 

```bash
nodetream run my-pipeline --target my-db
```

**Comment by ccloes** [comment]
I like this feature a lot.  Throw back to another system that I like a lot... ;)

**Comment by zprobst** [comment]
Stabilized / Released  in 0.9
